Sources/Parser/Parser+Statements.swift

```diff
@@ -118,6 +121,22 @@ extension Parser {
                        elseClauseStatements: elseClauseStatements)
   }
 
+  func parseDoCatchStatement() throws -> DoCatchStatement {
+    try consume(.do, or: .expectedStatement(at: latestSource))
+    let (doStatements, _) = try parseCodeBlock()
+    try consume(.catch, or: .expectedStatement(at: latestSource))
+    // Parse catch is Error
+    try consume(.is, or: .expectedStatement(at: latestSource))
+
+    guard let nextOpenBraceIndex = indexOfFirstAtCurrentDepth([.punctuation(.openBrace)]) else {
+      throw raise(.leftBraceExpected(in: "Do-catch Statement", at: latestSource))
+    }
+
+    let err = try parseExpression(upTo: nextOpenBraceIndex)
```

Comment:
Even though we don't have error types yet, I would use `parseType` here. Whatever error we will end up using will be special and unique for now, but will be a type nonetheless.

Reply:
Did you resolve this offline? What was the result of this discussion?

Reply:
Addressed in the latest commit.